evm/src/memory/memory_stark.rs

```diff
@@ -127,6 +127,11 @@ pub(crate) fn generate_first_change_flags_and_rc<F: RichField>(
         let address_changed =
             row[CONTEXT_FIRST_CHANGE] + row[SEGMENT_FIRST_CHANGE] + row[VIRTUAL_FIRST_CHANGE];
         row[INITIALIZE_AUX] = next_segment * address_changed * next_is_read;
+
+        let next_row = trace_rows[idx + 1].as_mut_slice();
+        if context_first_change || segment_first_change {
+            next_row[FIRST_OFFSET] = next_virt;
```

Comment:
Is there a reason why we're storing it in the next row and not the current one? (it's fine, I was just wondering)